### PR TITLE
Author page

### PR DESCRIPTION
**Pull request by soorajbala**

### CODE REVIEW

File: app/assets/stylesheets/layout/_author-page.scss [hunk 1]
```diff
@@ -0,0 +1,190 @@
+.author-detail-page {
+	@extend .container;
```
Comment (rvgpl) [inline app/assets/stylesheets/layout/_author-page.scss]: Don't use extends.

File: app/assets/stylesheets/layout/_author-page.scss [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+  box-shadow: 0 1px 6px 0 rgba($black, 0.1);
+  flex-direction: column;
+  margin-bottom: 20px;
+  @include breakpoint($pc){
+  	flex-direction: row;
+  	grid-column: span 2;
+  	margin-bottom: 0;
+  }
+	&__avatar {
+    @include aspect-ratio(18,19);
```
Comment (rvgpl) [inline app/assets/stylesheets/layout/_author-page.scss]: That is a weird aspect ratio.  Is this the zeplin mockup https://app.zeplin.io/project/5845366c39aa5361630e088f/screen/5a795f3ac0748dde1f9cdaed ?

File: app/assets/stylesheets/layout/_author-page.scss [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+	&__content {
+		flex: 0 1 65%;
+    padding: 30px;
+	}
+  
+	&__name {
+    font-family: var(--sansTypeface);
+	  font-size: 28px;
+	  font-weight: normal;
+	  color: #313131;
```
Comment (rvgpl) [inline app/assets/stylesheets/layout/_author-page.scss]: Color variable

File: app/assets/stylesheets/layout/_author-page.scss [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+	  text-align: left;
+	}
+
+	&__bio {
+    text-align: left;
+    font-family: var(--serifTypeface);
+	  font-size: 14px;
+	  font-weight: normal;
+	  line-height: 1.57;
+	  color: #636363;
```
Comment (rvgpl) [inline app/assets/stylesheets/layout/_author-page.scss]: Color variable

File: app/isomorphic/components/basic/story-card.js [hunk 5]
```diff
@@ -20,6 +20,7 @@ function StoryCard(props) {
 function StoryCardBgImage(props) {
   const author = {
     name: _.get(props.story, ['authors', 0, 'name'], props.story['author-name']),
+    slug: _.get(props.story, ['authors' , 0, 'id'] , '')
```
Comment (rvgpl) [inline app/isomorphic/components/basic/story-card.js]: I see in the above component the slug is `"slug": _.get(story, ['authors' , 0, 'id'] , story['author-id'])`, you can use the same here no? or I'm missing something

File: app/isomorphic/components/pages/author-page.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+              <figure className="author-loadmore-stories__story-image">
+				        <ResponsiveImage slug={story["hero-image-s3-key"]} metadata={story["hero-image-metadata"]}
+				          aspectRatio={[4,3]}
+				          defaultWidth={480} widths={[300,400,500]} sizes="(max-width: 500px) 98%, (max-width: 768px) 48%, 23%"
+				          imgParams={{auto:['format', 'compress'], fit:'max'}}/>
+				      </figure>
+			      </div>
+			      <div className="author-loadmore-stories__story__content">
+          	  <h3 className="author-loadmore-stories__story-heading">{story.headline}</h3>
+          	  {story['published-at'] && <p className="author-main-stories__story-published-at">
```
Comment (rvgpl) [inline app/isomorphic/components/pages/author-page.js]: Use a `div` it is not a paragraph.

File: app/assets/stylesheets/layout/_author-page.scss [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+
+.load-more-stories {
+  position: relative;
+  display: flex;
+  justify-content: flex-end;
+  font-size: 16px;
+  padding-bottom: 10px;
+  &__line {
+    background-color: inherit;
+    content: '';
```
Comment (rvgpl) [inline app/assets/stylesheets/layout/_author-page.scss]: It is not a pseudo element, no need for content

File: app/isomorphic/components/pages/author-page.js [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+  	const stories = props.data.stories;
+		const author = props.data.author;
+		const fields = "id,hero-image-s3-key,hero-image-metadata,headline,published-at";
+		return (
+			<div className="author-detail-page">
+				<section className="author-detail">
+				 <div className="author-detail__avatar">
+					 	<figure>
+					  	<img src={author['avatar-url']} />
```
Comment (rvgpl) [inline app/isomorphic/components/pages/author-page.js]: `alt` attribute yo

File: app/isomorphic/components/pages/author-page.js [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+					  </figure>
+				 </div>
+         <div className="author-detail__content">
+						<h1 className="author-detail__name">{author.name}</h1>
+						{ author.bio ?
```
Comment (rvgpl) [inline app/isomorphic/components/pages/author-page.js]: You can use `&&` to be make the code a bit clean. 
```{ author.bio && <div className="author-detail__bio" dangerouslySetInnerHTML={ {__html: author.bio }} />}```

File: app/isomorphic/components/pages/author-page.js [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+						<div className="author-detail__bio" dangerouslySetInnerHTML={ {__html: author.bio }} />
+						 : " " }
+						<div className="author-detail__social">
+						{ author['twitter-handle'] ?
```
Comment (rvgpl) [inline app/isomorphic/components/pages/author-page.js]: Same here also. Use `&&`

File: app/isomorphic/components/pages/author-page.js [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+						<div className="author-detail__social">
+						{ author['twitter-handle'] ?  
+						  <a href={author['twitter-handle']} target="_blank">
+						    <TwitterIcon />
+						  </a> : " " }
+						</div>
+					</div>
+				</section>
+				<section className="author-page-dfp">
+				  <DfpAd adtype="Mrec" />
```
Comment (rvgpl) [inline app/isomorphic/components/pages/author-page.js]: You need to pass targetting parameters, check with @asimqt

File: app/isomorphic/components/story-page-components/story-header-card.js [hunk 12]
```diff
@@ -86,7 +86,9 @@ class StoryHeaderCard extends React.Component {
           <Author author={{
               "name": _.get(this.state.story, ['authors', 0, 'name'], this.state.story['author-name']),
               "image": this.state.story['author-image'],
-              "date": this.state.story['first-published-at']}} />
+              "date": this.state.story['first-published-at'],
+              "slug": this.state.story['author-id']
```
Comment (rvgpl) [inline app/isomorphic/components/story-page-components/story-header-card.js]: Why is this not picking up `authors`?

File: app/assets/stylesheets/layout/_author-page.scss (hunk 1 above)
Comment (soorajbala) [reply app/assets/stylesheets/layout/_author-page.scss]: yes even i don't want to use that but for all components it is used like that only, so i followed same here.

File: app/assets/stylesheets/layout/_author-page.scss (hunk 2 above)
Comment (soorajbala) [reply app/assets/stylesheets/layout/_author-page.scss]: yes

Comment (soorajbala) [reply app/assets/stylesheets/layout/_author-page.scss]: chnaged to 1:1

File: app/isomorphic/components/story-page-components/story-header-card.js (hunk 12 above)
Comment (soorajbala) [reply app/isomorphic/components/story-page-components/story-header-card.js]: done

File: app/isomorphic/components/pages/author-page.js (hunk 11 above)
Comment (soorajbala) [reply app/isomorphic/components/pages/author-page.js]: checked with asim because it is author page for Mrec we dont want to pass targettig params

**Pull request merged**